src/io/ResultWriter.cpp

```diff
@@ -1,89 +1,111 @@
-			<< std::setw(_writePrecision+15) << "beta_trans"
-			<< std::setw(_writePrecision+15) << "beta_rot"
 			<< std::setw(_writePrecision+15) << "c_v"
 			<< std::setw(_writePrecision+15) << "N"
 			<< endl;
+		resultStream.close();
 	}
 }
 
 void ResultWriter::endStep(ParticleContainer *particleContainer, DomainDecompBase *domainDecomp, Domain *domain,
                            unsigned long simstep) {
 
-	// Writing of cavities now handled by CavityWriter
+	const uint64_t globalNumMolecules = domain->getglobalNumMolecules(true, particleContainer, domainDecomp);
 
-	unsigned long globalNumMolecules = domain->getglobalNumMolecules(true, particleContainer, domainDecomp);
-	double cv = domain->cv();
+	_uPot_acc += domain->getGlobalUpot();
+	_p_acc += domain->getGlobalPressure();
 
-	_U_pot_acc->addEntry(domain->getGlobalUpot());
-	_p_acc->addEntry(domain->getGlobalPressure());
-	if((domainDecomp->getRank() == 0) && (simstep % _writeFrequency == 0)){
-		_resultStream << std::setw(10) << simstep << std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _simulation.getSimulationTime()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalUpot()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _U_pot_acc->getAverage()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalPressure()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << _p_acc->getAverage()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalBetaTrans()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << domain->getGlobalBetaRot()
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << cv
-			<< std::setw(_writePrecision+15) << std::scientific << std::setprecision(_writePrecision) << globalNumMolecules
-			<< endl;
+	double uKinLocal = 0.0F;
+	double uKinGlobal = 0.0F;
+	double uKinTransLocal = 0.0F;
+	double uKinTransGlobal = 0.0F;
+	double uKinRotLocal = 0.0F;
+	double uKinRotGlobal = 0.0F;
+	#if defined(_OPENMP)
+	#pragma omp parallel reduction(+:uKinLocal,uKinTransLocal,uKinRotLocal)
+	#endif
+	{
+		for (auto moleculeIter = particleContainer->iterator(ParticleIterator::ONLY_INNER_AND_BOUNDARY);
+			 moleculeIter.isValid(); ++moleculeIter) {
+			uKinLocal += moleculeIter->U_kin();
+			uKinTransLocal += moleculeIter->U_trans();
+			uKinRotLocal += moleculeIter->U_rot();
+		}
 	}
-}
+#ifdef ENABLE_MPI
+    MPI_Allreduce(&uKinLocal, &uKinGlobal, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+	MPI_Allreduce(&uKinTransLocal, &uKinTransGlobal, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
+	MPI_Allreduce(&uKinRotLocal, &uKinRotGlobal, 1, MPI_DOUBLE, MPI_SUM, MPI_COMM_WORLD);
```

Comment:
Can we use `DomainDecompMPIBase.h::getCommunicator()` here instead of `MPI_COMM_WORLD`? In my particular case I'm saying this because when I have multiple MD instances running with MaMiCo, each of the instances have separate communicators. But generally, won't the code be better if we use the communicator? Or is it unnecessary for result writing anyway?

Reply:
To be honest, I am not exactly sure what the difference is. But I think, that it does not really matter for the ResultWriter. I changed it to `DomainDecompMPIBase.h::getCommunicator()`.